### PR TITLE
Untie from Android System Auto Sync settings

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/34136

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Setup Brave Sync on Android
2. Exit browser
3. Turn off system Auto Sync settings, see [Why won’t my data Sync on my Android device?](https://support.brave.com/hc/en-us/articles/360047642371-Sync-FAQ)  
4. Launch Brave Browser
5. Expected not to see the informer as on screenshots from https://github.com/brave/brave-core/pull/7316 and have green statuses at brave://sync-internals

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java [hunk 1]
```diff
@@ -791,18 +809,24 @@ private enum SyncInputType { NEW, JOIN }
...
+                                    setAppropriateView();
+                                }
+                            } else {
+                                // TODO(AlexeyBarabash): consider to have error code if there will
+                                // be more than only one case of failure
+                                onSyncError(
+                                        getResources()
+                                                .getString(
+                                                        R.string
+                                                                .brave_sync_joining_deleted_account));
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java:826]: presubmit format likes [Mayakovsky](https://en.wikipedia.org/wiki/Vladimir_Mayakovsky#/media/File:%D0%9A%D0%BE%D0%BB%D1%85%D0%BE%D0%B7%D0%BE%D0%BD%D1%8C_%D1%8D%D1%80%D1%8F%D1%84_1940-06_37%D1%81%D1%82%D1%80.jpg)

**Review by samartnik — APPROVED** [review]
lgtm

**Comment by samartnik** [comment]
don't we need to remove this patch too? https://github.com/brave/brave-core/commit/924a3dd25a71db833c584106f9af50a4d5cb90ad#diff-dd98e4f5a3c59f028e7c2d79502a1e74632864e42490c840b266bdaeb36c9116R53

**Comment by AlexeyBarabash** [comment]
@samartnik 

> don't we need to remove this patch too? [924a3dd#diff-dd98e4f5a3c59f028e7c2d79502a1e74632864e42490c840b266bdaeb36c9116R53](https://github.com/brave/brave-core/commit/924a3dd25a71db833c584106f9af50a4d5cb90ad#diff-dd98e4f5a3c59f028e7c2d79502a1e74632864e42490c840b266bdaeb36c9116R53)

We need. Thanks, I missed that.

Fixed at https://github.com/brave/brave-core/pull/20854/commits/e86a9da90dd4c2b8bad477cecef017e5a613fe21 .

**Review by mkarolin — APPROVED** [review]
`patches`++
`strings`++

**Review by emerick — APPROVED** [review]

**Review by tapanmodh — APPROVED** [review]
lgtm

File: patches/chrome-android-java-AndroidManifest.xml.patch [hunk 2]
```diff
@@ -54,7 +54,7 @@ index fba6543bde641c1e29cbac01914000c75495cf2b..da9879111318168f381ffc3727b73a35
  
 +<!--
      <uses-permission android:name="com.google.android.apps.now.CURRENT_ACCOUNT_ACCESS" />
-+--> <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE"/> <uses-permission android:name="android.permission.READ_SYNC_SETTINGS"/>
++--> <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE"/>
```
Comment (bridiver) [inline patches/chrome-android-java-AndroidManifest.xml.patch:57]: can't we get rid of this line entirely if we put this permission inside `{% include "//brave/android/java/AndroidManifest_user_permissions.xml" %}`

**Pull request merged**